ouroboros-network/src/Ouroboros/Network/TxSubmission/Inbound.hs

```diff
@@ -324,8 +332,15 @@ txSubmissionInbound _tracer maxUnacked mpReader mpWriter _version =
             bufferedTxs3 = forceElemsToWHNF $ bufferedTxs2 <>
                                (Map.fromList (zip live (repeat Nothing)))
 
+        let !collected = length txs
+        traceWith tracer $
+          TraceTxSubmissionCollected collected
+
+        txidsAccepted <- mempoolAddTxs txsReady
 
-        _writtenTxids <- mempoolAddTxs txsReady
+        let !accepted = length txidsAccepted
+        traceWith tracer $ TraceTxSubmissionAccepted accepted
+        traceWith tracer $ TraceTxSubmissionRejected (collected - accepted)
```

Comment:
Maybe you'd like to only emit this when `collected > accepted`?

Reply:
Excellent idea.  Done!